<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>480</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>800</width>
    <height>480</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>800</width>
    <height>543</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Smart Clock</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QStackedWidget" name="stackedWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="clockPage">
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QLabel" name="timeLabel">
          <property name="font">
           <font>
            <family>Arial</family>
            <pointsize>72</pointsize>
           </font>
          </property>
          <property name="text">
           <string>00:00:00</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="dateLabel">
          <property name="font">
           <font>
            <family>Arial</family>
            <pointsize>24</pointsize>
           </font>
          </property>
          <property name="text">
           <string>Monday, January 1, 2024</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="newsPage">
       <layout class="QVBoxLayout" name="verticalLayout_8">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QLabel" name="newsHeaderLabel">
            <property name="font">
             <font>
              <pointsize>16</pointsize>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>Latest News</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QScrollArea" name="newsScrollArea">
            <property name="widgetResizable">
             <bool>true</bool>
            </property>
            <widget class="QWidget" name="scrollAreaWidgetContents">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>0</y>
               <width>760</width>
               <height>288</height>
              </rect>
             </property>
             <layout class="QVBoxLayout" name="newsLayout"/>
            </widget>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_7">
            <item>
             <widget class="QPushButton" name="refreshNewsButton">
              <property name="text">
               <string>Refresh News</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="nextSourceButton">
              <property name="text">
               <string>Next Source</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="radioPage">
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <widget class="QLabel" name="radioHeaderLabel">
          <property name="font">
           <font>
            <pointsize>16</pointsize>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>Radio Stations</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QListWidget" name="radioListWidget">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>200</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="radioControlsGroup">
          <property name="title">
           <string>Radio Controls</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_7">
           <item>
            <layout class="QHBoxLayout" name="radioControlsLayout">
             <item>
              <widget class="QPushButton" name="playButton">
               <property name="minimumSize">
                <size>
                 <width>80</width>
                 <height>30</height>
                </size>
               </property>
               <property name="text">
                <string>Play</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="stopButton">
               <property name="minimumSize">
                <size>
                 <width>80</width>
                 <height>30</height>
                </size>
               </property>
               <property name="text">
                <string>Stop</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QGroupBox" name="volumeGroup">
             <property name="title">
              <string>Volume</string>
             </property>
             <layout class="QHBoxLayout" name="horizontalLayout">
              <item>
               <widget class="QSlider" name="volumeSlider">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="volumeLabel">
                <property name="minimumSize">
                 <size>
                  <width>30</width>
                  <height>0</height>
                 </size>
                </property>
                <property name="text">
                 <string>50%</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="alarmPage">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_10">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <property name="sizeConstraint">
           <enum>QLayout::SetMaximumSize</enum>
          </property>
          <item>
           <widget class="QLabel" name="alarmHeaderLabel">
            <property name="font">
             <font>
              <pointsize>16</pointsize>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>Alarms      Week Days                                                   Weekends</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <property name="sizeConstraint">
             <enum>QLayout::SetMaximumSize</enum>
            </property>
            <item>
             <widget class="QListWidget" name="alarmWeekDayListWidget"/>
            </item>
            <item>
             <widget class="QListWidget" name="alarmWeekEndListWidget"/>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QCheckBox" name="alarmCheckBox">
            <property name="text">
             <string>Alarm Enabled</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="stopAlarmButton">
            <property name="text">
             <string>Start Alarm</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="trainsPage">
       <layout class="QVBoxLayout" name="verticalLayout_6">
        <item>
         <widget class="QLabel" name="trainsHeaderLabel">
          <property name="font">
           <font>
            <pointsize>16</pointsize>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>London Train Status</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QScrollArea" name="trainsScrollArea">
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="scrollAreaWidgetContents_2">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
             <width>762</width>
             <height>292</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="trainsLayout"/>
          </widget>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="refreshTrainsButton">
          <property name="text">
           <string>Refresh Status</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="weatherPage">
       <layout class="QVBoxLayout" name="verticalLayout_9">
        <item>
         <widget class="QLabel" name="weatherLabel">
          <property name="font">
           <font>
            <pointsize>16</pointsize>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>Weather Forecast</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QWidget" name="weatherContainer" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="refreshWeatherButton">
          <property name="text">
           <string>Refresh</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="QWidget" name="navigationWidget" native="true">
      <layout class="QHBoxLayout" name="navigationLayout">
       <item>
        <widget class="QPushButton" name="clockButton">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>40</height>
          </size>
         </property>
         <property name="text">
          <string>Clock</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="newsButton">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>40</height>
          </size>
         </property>
         <property name="text">
          <string>News</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="radioButton">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>40</height>
          </size>
         </property>
         <property name="text">
          <string>Radio</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="alarmButton">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>40</height>
          </size>
         </property>
         <property name="text">
          <string>Alarm</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="trainsButton">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>40</height>
          </size>
         </property>
         <property name="text">
          <string>Trains</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="weatherButton">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>40</height>
          </size>
         </property>
         <property name="text">
          <string>Weather</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
